### PR TITLE
Map ruler 

### PR DESCRIPTION
**Pull request by EugeniaSimonova**

### CODE REVIEW

File: interface-elements-for-desktop/articles/map/rulers.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Rulers
+You can use rulers to measure distances and areas on a map.
+
+| Distance Ruler | Area Ruler | 
+|:---:|:---:|
+| ![Map Distance Ruler](../../images/map-ruler-distance.png) | ![Map Area Ruler](../../images/map-ruler-area.png)| 
+
+## Measure Distances
+
+Click  the ![Distance ruler button](../../images/map-ruler-distance-icon.png) button to create a distance ruler. This ruler calculates the distance between two or more points on a map. The total distance is shown in a map callout. Note that measurements ignore changes in elevation.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/map/rulers.md:15]:
```suggestion
Click  the ![Distance ruler button](../../images/map-ruler-distance-icon.png) button to create a ruler. This ruler calculates the distance between two or more points on a map. The total distance is shown in a map callout. Note that measurements ignore changes in elevation.
```

File: interface-elements-for-desktop/articles/map/rulers.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+![Create distance ruler](../../images/map-distance-ruler.gif) 
+
+Click an intermediate point on the ruler to see the distance to this point.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/map/rulers.md:19]: What is the difference between 'a point' and 'an intermediate point'?

File: interface-elements-for-desktop/articles/map/rulers.md (hunk 1 above)
Comment (dirk-pieterse) [reply interface-elements-for-desktop/articles/map/rulers.md:15]: "This ruler calculates the distance between two or more points on a map" Do you use the ruler to measure the distance, or does the ruler calculate the distance itself?

File: interface-elements-for-desktop/articles/map/rulers.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+![Show the distance to an intermediate point](../../images/map-ruler-intermediate-point.gif) 
+
+## Measure Areas 
+Click ![Area ruler button](../../images/map-ruler-area-icon.png) button to create an area ruler. This ruler calculates the area inside a polygon. The result is shown in a map callout.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/map/rulers.md:24]: "This ruler calculates the area inside a polygon" The same question here

File: interface-elements-for-desktop/articles/map/rulers.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+![Create distance ruler](../../images/map-area-ruler.gif) 
+## Configure Rulers
+
+After you add a ruler to a map, you can perform the following actions:
+
+* **Add a ruler point**.  Click the position on the ruler where you wish to add a point.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/map/rulers.md:31]:
```suggestion
* **Add a ruler point**.  Click the position on the ruler where you want to add a point.
```

File: interface-elements-for-desktop/articles/map/rulers.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    ![ Add a ruler point](../../images/map-ruler-add-point.gif) 
+
+* **Change the ruler polyline**. Click and drag the point to a new location.
+
+    ![ Move a ruler point](../../images/map-ruler-move-point.gif)
+
+* **Delete a ruler point**. Double-click the ruler point. You can delete a point when the ruler contains more than two points. You can remove only the entire ruler if it contains two points.
```
Comment (dirk-pieterse) [inline interface-elements-for-desktop/articles/map/rulers.md:39]:
```suggestion
* **Delete a ruler point**. Double-click the ruler point. You can delete a point when the ruler contains more than two points. You can remove the entire ruler if it contains only two points.
```

**Review by dirk-pieterse — APPROVED** [review]

File: interface-elements-for-desktop/articles/map/rulers.md (hunk 1 above)
Comment (EugeniaSimonova) [reply interface-elements-for-desktop/articles/map/rulers.md:15]: resolved in teams

[8:59 AM] Eugenia Simonova (DevExpress)
    Good morning, Dirk (smile),
about https://github.com/DevExpress/dotnet-eud/pull/114

1.

"This ruler calculates the distance between two or more points on a map" Do you use the ruler to measure the distance, or does the ruler calculate the distance itself?

You click a map to set points and a ruler shows the distance value between these points. So, I would say that the ruler calculates the distance.

The same with area rulers. 


2.


"Click an intermediate point on the ruler to see the distance to this point."

What is the difference between 'a point' and 'an intermediate point'?


I meant:  intermediate point =not the first point and not the last point. 


If you click the first or the last point, nothing happens.


(1 liked)

**Pull request closed without merging**